tests/string_test.ts

```diff
@@ -0,0 +1,181 @@
+  const v = await client.get("dest");
+  assertEquals(v, "bab"); // 01100010 01100001 01100000
+});
+
+test("bitpos", async () => {
+  await client.set("key", "2"); // 00110010
+  assertEquals(await client.bitpos("key", 0), 0);
+  assertEquals(await client.bitpos("key", 1), 2);
+});
+
+test("decr", async () => {
+  const rep = await client.decr("key");
+  assertEquals(rep, -1);
+  assertEquals(await client.get("key"), "-1");
+});
+
+test("decby", async () => {
+  const rep = await client.decrby("key", 101);
+  assertEquals(rep, -101);
+  assertEquals(await client.get("key"), "-101");
+});
+
+test("getWhenNil", async () => {
+  const hoge = await client.get("none");
+  assertEquals(hoge, undefined);
+});
+
+test("getbit", async () => {
+  await client.set("key", "3"); // 00110011
+  assertEquals(await client.getbit("key", 0), 0);
+  assertEquals(await client.getbit("key", 2), 1);
+});
+
+test("getrange", async () => {
+  await client.set("key", "Hello world!");
+  const v = await client.getrange("key", 6, 10);
+  assertEquals(v, "world");
+});
+
+test("getset", async function testGetSet() {
+  await client.set("key", "val");
+  const v = await client.getset("key", "lav");
+  assertEquals(v, "val");
+  assertEquals(await client.get("key"), "lav");
+});
+
+test("incr", async () => {
+  const rep = await client.incr("key");
+  assertEquals(rep, 1);
+  assertEquals(await client.get("key"), "1");
+});
+
+test("incrby", async () => {
+  const rep = await client.incrby("key", 101);
+  assertEquals(rep, 101);
+  assertEquals(await client.get("key"), "101");
+});
+
+test("incrbyfloat", async () => {
+  await client.set("key", "2.1");
+  const v = await client.incrbyfloat("key", 0.5);
+  assertEquals(v, "2.6");
+  assertEquals(await client.get("key"), "2.6");
+});
+
+test("mget", async () => {
+  await client.set("key1", "val1");
+  await client.set("key2", "val2");
+  await client.set("key3", "val3");
+  const v = await client.mget("key1", "key2", "key3");
+  assertEquals(v, ["val1", "val2", "val3"]);
+});
+
+test("mset", async () => {
+  const rep = await client.mset("key1", "foo", "key2", "bar", "key3", "baz");
+  assertEquals(rep, "OK");
+  assertEquals(await client.get("key1"), "foo");
+  assertEquals(await client.get("key2"), "bar");
+  assertEquals(await client.get("key3"), "baz");
+});
+
+test("msetnx", async () => {
+  const rep1 = await client.msetnx("key1", "foo", "key2", "bar");
+  assertEquals(rep1, 1); // All the keys were set.
+  const rep2 = await client.msetnx("key2", "baz", "key3", "qux");
+  assertEquals(rep2, 0); // No key was set.
+  assertEquals(await client.get("key1"), "foo");
+  assertEquals(await client.get("key2"), "bar");
+  assertEquals(await client.get("key3"), undefined);
+});
+
+// This test is a bit slow.
+test("psetex #slow", async () => {
+  const rep = await client.psetex("key1", 1000, "test");
+  assertEquals(rep, "OK");
+  assertEquals(await client.get("key1"), "test");
+  await sleep(1000);
```

Comment:
Please avoid consuming real time in tests. It doesn't need to check redis behaviour itself.